python/tests/test_writer.py

```diff
@@ -0,0 +1,56 @@
+import contextlib
+import json
+from tempfile import TemporaryFile
+
+from mcap.mcap0.writer import CompressionType, Writer
+
+
+@contextlib.contextmanager
+def generate_sample_data(compression: CompressionType):
+    file = TemporaryFile("w+b")
+    writer = Writer(file, compression=compression)
+    writer.start(profile="x-json", library="test")
+    schema_id = writer.register_schema(
+        name="sample",
+        encoding="jsonschema",
+        data=json.dumps(
+            {
+                "type": "object",
+                "properties": {
+                    "sample": {
+                        "type": "string",
+                    }
+                },
+            }
+        ).encode(),
+    )
+
+    channel_id = writer.register_channel(
+        schema_id=schema_id,
+        topic="sample_topic",
+        message_encoding="json",
+    )
+
+    writer.add_message(
+        channel_id=channel_id,
+        log_time=0,
+        data=json.dumps({"sample": "test"}).encode("utf-8"),
+        publish_time=0,
+    )
+
+    writer.finish()
+    file.seek(0)
+
+    yield file
```

Comment:
should this be wrapped in `with TemporaryFile(...)`? or is that only for NamedTemporaryFile?

Reply:
I think this one will clean itself up automatically:

```Return a [file-like object](https://docs.python.org/3/glossary.html#term-file-like-object) that can be used as a temporary storage area. The file is created securely, using the same rules as [mkstemp()](https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp). It will be destroyed as soon as it is closed (including an implicit close when the object is garbage collected).```